sql/expression/function/arraylength.go

```diff
@@ -57,19 +57,27 @@ func (f *ArrayLength) WithChildren(children ...sql.Expression) (sql.Expression,
 
 // Eval implements the Expression interface.
 func (f *ArrayLength) Eval(ctx *sql.Context, row sql.Row) (interface{}, error) {
-	if t := f.Child.Type(); !sql.IsArray(t) && t != sql.JSON {
+	t := f.Child.Type()
+	if !sql.IsArray(t) && !sql.IsJSON(t) {
 		return nil, nil
 	}
 
 	child, err := f.Child.Eval(ctx, row)
 	if err != nil {
 		return nil, err
 	}
-
 	if child == nil {
 		return nil, nil
 	}
 
+	if val, ok := child.(sql.JSONValue); ok {
```

Comment:
Same comment here. This function is not part of MySQL and has poorly defined semantics. Kill it.